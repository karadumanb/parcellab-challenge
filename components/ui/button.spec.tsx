import { render } from '@testing-library/react'
import React from 'react'
import { Button } from './button'

describe('Button', () => {
  it('default button snapshot', () => {
    const { asFragment } = render(<Button>Default Button</Button>)

    expect(asFragment()).toMatchSnapshot()
  })

  it('secondary button snapshot', () => {
    const { asFragment } = render(<Button variant="secondary">Secondary Button</Button>)

    expect(asFragment()).toMatchSnapshot()
  })

  it('as child button snapshot', () => {
    const { asFragment } = render(
      <Button asChild>
        <span>Child Button</span>
      </Button>
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
