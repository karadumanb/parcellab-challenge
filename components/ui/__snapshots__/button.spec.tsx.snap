// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button as child button snapshot 1`] = `
<DocumentFragment>
  <span
    class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
  >
    Child Button
  </span>
</DocumentFragment>
`;

exports[`Button default button snapshot 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
  >
    Default Button
  </button>
</DocumentFragment>
`;

exports[`Button secondary button snapshot 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-secondary text-secondary-foreground hover:bg-secondary/80 h-10 px-4 py-2"
  >
    Secondary Button
  </button>
</DocumentFragment>
`;
